tests/markdown_linkReference/title.md

```diff
@@ -0,0 +1,25 @@
+[ref]: https://example.com (bar)
+[other-ref]: https://example.com (Shakespeare's "Romeo and Juliet" is a famous play)
+
+[a](https://example.com "\"")
```

Comment:
These `[a]` test cases should be in `tests/markdown_link` as they're `link`s instead of `linkReference`s and we need to ensure the following cases are fixed:

**Prettier pr-3481**
[Playground link](https://deploy-preview-3481--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEBtAhgXQBQAsYwAOAzkgPRlwAe6AtoQDZwB0ktABADoie8gDkASm7CoGHPiKkK1OoxZt2-bryFDOYrHgIlylGvSasIHbN2WdBVkABoQEQjACW0YslDoATp4gB3AApeCG4o6ABuEE4AJrYg6MQwyABm6AzEcHYARp7oYADWcDAAyoS5TlAA5sgwngCuGSBREGDJqel25emeMP45FbTorWkNAFbEVABCOfmFRXRwADLlcEPt9rVEGwBMqw2lnl3IIAOeeU2+ULGEnuUwAOrRMLjIABwADHbXEOl3OYRH1zgXTCKzsnjgAEdak5wb10P1BkgUsM7OlaE5qnUGsRyhUmABFWoQeC7OwwdCZB5RJ7ILZknJOBi4gDCJgGRyg0FBIFq6QAKhSQsj0gBfEVAA)
```sh
--parser markdown
```

**Input:**
```markdown
[a](https://example.com "\"')")
[a](https://example.com '"\')')
[a](https://example.com ("'\)))
```

**Output:**
```markdown
[a](https://example.com "$1\"')")
[a](https://example.com "$1\"')")
[a](https://example.com "$1\"')")

```